tests/main/component/task.yaml

```diff
@@ -37,15 +42,21 @@ execute: |
       stat "$comp_inst_path"
 
       # Component is mounted (note that we need to escape the "+" in the path)
-      # x1 is the snap revision
-      symlink=$SNAP_MOUNT_DIR/snap-with-comps/components/x1/comp1
       mnt_point=$SNAP_MOUNT_DIR/snap-with-comps/components/mnt/comp1/${comp_rev}
       mount | MATCH "^${comp_inst_path/+/\\+} on ${mnt_point} .*"
       # And symlinked
       readlink "$symlink" | MATCH "\\.\\./mnt/comp1/$comp_rev"
       readlink -f "$symlink" | MATCH "$mnt_point"
       # and is seen from snap app
       snap-with-comps.test
+
+      # Old component is not mounted and has been removed
+      if [ -n "$prev_comp_rev" ]; then
+          prev_mnt_point=$SNAP_MOUNT_DIR/snap-with-comps/components/mnt/comp1/${prev_comp_rev}
+          prev_comp_inst_path=/var/lib/snapd/snaps/snap-with-comps+comp1_${prev_comp_rev}.comp
+          mount | not MATCH "^${prev_comp_inst_path/+/\\+} on ${prev_mnt_point} .*"
+          not stat "$prev_comp_inst_path"
```

Comment:
do we need to check whether directories above the mountdir where removed as well?

Reply:
Check added